browser/brave_startup_data.cc

```diff
@@ -0,0 +1,15 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_startup_data.h"
+
+#include "brave/browser/browser_context_keyed_service_factories.h"
+
+#if defined(OS_ANDROID)
+void BraveStartupData::PreProfilePrefServiceInit() {
+  StartupData::PreProfilePrefServiceInit();
+  brave::EnsureBrowserContextKeyedServiceFactoriesBuilt();
```

Comment:
Android need to call `brave::EnsureBrowserContextKeyedServiceFactoriesBuilt();` here otherwise it would crash at startup due to accessing unregistered ads preferences.

Reply:
this can be accomplished with a chromium src override for `ChromeBrowserMainExtraPartsProfiles` in startup_data.cc and then add the static method to `BraveBrowserMainExtraPartsProfiles`

Reply:
Thanks a lot, addressed in https://github.com/brave/brave-core/pull/3905/commits/f28e9135d6f5df6d672d6312a9425c323a9e643e